dependabot_time_to_merge.rb

```diff
@@ -1,83 +1,115 @@
+
+    # There are a few PRs that don't have the expected structure.
+    # We can ignore those since the number is low (for example, from 1040 PRs raised 33 have a different structure)
+    # and it won't have a big impact on the final results
+    return nil if details.nil?
 
-def dependabot_prs
-  client.search_issues("is:pr user:alphagov is:closed author:app/dependabot archived:false", per_page: 100)
-  
-  last_response = client.last_response
-
-  return [] if last_response.data.items.empty?
-
-  pulls = []
-  sleep_time = 60 # GitHub's Search API applies strict secondary rate-limiting
-
-  # GitHub Search API has a limit of 1000 results,
-  # so we should only loop through 100-results-per-page 10 times.
-  # https://docs.github.com/en/rest/search#about-the-search-api
-  10.times do |i|
-    puts "Fetching next page of results... (API call ##{i+1})\nSleeping #{sleep_time} seconds to avoid rate-limiting..."
-    pulls << last_response.data.items
-    break if last_response.rels[:next].nil?
-    sleep sleep_time
-    last_response = last_response.rels[:next].get
+    [details[1], details[2]]
   end
-  pulls.flatten
-end
 
-def govuk_prs
-  @govuk_prs ||=
-    dependabot_prs.flatten.select do |pr|
-      govuk_repos.any? { |repo| pr.repository_url.include?(repo) }
-    end
-end
+  def get_repo_prs(repo)
+    client.list_issues(repo, { state: 'all', labels: 'dependencies' })
+  rescue Octokit::NotFound
+    []
+  end
 
-def filter_by_date(from, to)
-  govuk_prs.flatten.select do |pr|
-    pr.pull_request["merged_at"].between?(Time.parse(from), Time.parse(to)) if pr.pull_request["merged_at"]
+  def dependabot_history_per_repo(repo)
+    repo_dependabot_prs = {}
+
+    get_repo_prs(repo).each do |pr|
+      dependency_name, from_version = get_dependency_name_and_version(pr.title)
+
+      next if dependency_name.nil?
+
+      repo_dependabot_prs[dependency_name] = [] if repo_dependabot_prs[dependency_name].nil?
+      repo_dependabot_prs[dependency_name] << {
+        created_at: pr.created_at,
+        merged_at: pr.pull_request['merged_at'],
+        closed_at: pr.closed_at,
+        from_version: from_version,
+      }
+    end
+    repo_dependabot_prs
   end
-end
 
-def time_to_merge(from, to)
-  @time_to_merge = []
-  filter_by_date(from, to).each do |pr|
-    days_to_merge = (pr.pull_request["merged_at"] - pr.created_at).to_i / (24 * 60 * 60)
-    puts "Created at: #{pr.created_at}, merged at: #{pr.pull_request["merged_at"]}, time to merge: #{days_to_merge} days, url: #{pr.url}"
-    @time_to_merge << days_to_merge
+  def get_repo_metrics(repo, from, to)
+    # Get the insights for each repository
+    dependabot_prs = dependabot_history_per_repo(repo)
+
+    total_opened_prs = 0
+    time_to_merge = []
+    time_since_open = []
+
+    dependabot_prs.keys.each do |dependency|
```

Comment:
I _think_ you can make this more readable like:

```suggestion
    dependabot_prs.each do |_dependency, prs|
```

And then just call `opened_prs = prs.filter { |pr| pr[:created_at].between?(from, to)}`, etc.